podtailer/podtailer.go

```diff
@@ -81,10 +82,30 @@ loop:
 		case pod := <-podWatcher.Pods():
 			watcher, err := pt.watcherForPod(pod, pt.config.ContainerName, podWatcher)
 			if err != nil {
-				// This shouldn't happen, since we check for configuration errors
-				// before actually setting up the watcher
-				logrus.WithError(err).Error("Error setting up watcher")
-				continue loop
+				retries := 0
+				// sometimes the pod isn't ready - for example, its log directory might not be
+				// on disk yet - back off and retry on incremental intervals before giving up
+				for true {
```

Comment:
FYI: it can just be `for {` (the `true` is redundant)

Reply:
But why not consolidate this line to `for retries := 0; retries < maxRetries; retries++ {` ?